### PR TITLE
Allow to sort the XML & fixes

### PR DESCRIPTION
**Pull request by hoijui**
The sorting is probably quite buggy,
but for a first trial with a simple SVG from Inkspace,
it does what I want and the SVG is still valid
and looks the same.

### CODE REVIEW

**Comment by hoijui** [comment]
I also have the code for xml-pretty ready already.

File: src/document.rs [hunk 1]
```diff
@@ -48,6 +54,90 @@ impl Declaration {
...
 }
 
+pub struct ElementAndContext<'a> {
+    pre: Vec<&'a Node>,
+    elem: &'a Node,
+    attrs: &'a IndexMap<QName, String>,
+    elem_val: &'a ElementValue,
+}
+
+pub fn node2ordinal(node: &Node) -> u8 {
```
Comment (bbqsrc) [inline src/document.rs]: Make this a to_ordinal method on Node itself.

File: src/document.rs [hunk 2]
```diff
@@ -70,6 +160,88 @@ impl Document {
...
+        let mut new_node_value = node_value.clone();
+        if let NodeValue::Element(ref mut e) = new_node_value {
+            e.children = self.sort_nodes(&e.children);
+        }
+        new_node_value
+    }
+
+    fn sort_attrs(&mut self) {
+        for el_attrs in self.attrs.values_mut() {
+            el_attrs.sort_by(ord_q_name);
```
Comment (bbqsrc) [inline src/document.rs]: `ord_q_name` is simple enough it can just be a closure.

File: src/document.rs [hunk 3]
```diff
@@ -70,6 +160,88 @@ impl Document {
...
+                    panic!("DocumentType found inside an element; that should never be the case")
+                }
+            }
+        }
+        if has_text && !elems.is_empty() {
+            println!(
+                "Element contains both element(s) and text; \
+            we thus consider all elements to be inline-elements \
+            (think of the *bold* element in HTML), and thus we will not sort them."
+            );
```
Comment (bbqsrc) [inline src/document.rs]: This should be logging not a println; if we don't already have the `tracing` dependency, let's add it.

**Review by bbqsrc — CHANGES_REQUESTED** [review]
A few things I'd like changed if you don't mind. 😄

**Comment by hoijui** [comment]
Should be done now

**Comment by bbqsrc** [comment]
Thanks :)

**Pull request merged**